pytissueoptics/scene/tree/treeConstructor/binary/binaryNodeSplitter.py

```diff
@@ -82,3 +82,56 @@ def _searchMinSAH(self, nodeBbox, polygons, splitAxis, aMin, step):
 
         splitLine = aMin + lowestIndexSAH * step
         return splitLine, minSAH
+
+
+class ShrankBoxSAHNodeSplitter(NodeSplitter):
+    def __init__(self, polygonCounter: PolygonCounter = None,
+                 nbOfSplitPlanes: int = 20, splitCostPercentage: float = 0.1):
+        super().__init__(polygonCounter)
+        self._nbOfSplitPlanes = nbOfSplitPlanes
+        self._splitCostPercentage = splitCostPercentage
+
+    def split(self, splitAxis: str, nodeBbox: BoundingBox, polygons: List[Polygon]) -> SplitNodeResult:
+        polygonsBbox = getPolygonsBbox(polygons)
+        newBounds = nodeBbox.copy()
+        newBounds.shrinkTo(polygonsBbox)
+        newBounds.update(splitAxis, "min", newBounds.getAxisLimit(splitAxis, "min")-0.1)
+        newBounds.update(splitAxis, "max", newBounds.getAxisLimit(splitAxis, "max")+0.1)
+        aMin, aMax = newBounds.getAxisLimits(splitAxis)
+        step = newBounds.getAxisWidth(splitAxis) / (self._nbOfSplitPlanes + 1)
+
+        nodeSAH = nodeBbox.getArea() * len(polygons)
+        splitLine, minSAH = self._searchMinSAH(newBounds, polygons, splitAxis, aMin, step)
+        splitCost = self._splitCostPercentage * nodeSAH
+
+        if minSAH + splitCost < nodeSAH:
+            stopCondition = False
+
+        else:
+            stopCondition = True
+
+        polygonGroups = self._polygonCounter.count(splitLine, splitAxis, polygons)
+        groupBbox = self._getNewChildrenBbox(newBounds, splitAxis, splitLine)
+        return SplitNodeResult(stopCondition, splitAxis, splitLine, groupBbox, polygonGroups)
```

Comment:
Duplicated code at HardSAH.
Maybe create an SAH abstract class or something. Or inherit from HardSAH if this is a variation of the latter.

Reply:
resolved with commit fd6db356bfb184947a23962841f3a516446f088c